app/controllers/concerns/recovery_code_concern.rb

```diff
@@ -0,0 +1,12 @@
+module RecoveryCodeConcern
+  delegate :active_profile, to: :current_user
+
+  def create_new_code
+    if active_profile.present?
+      Pii::ReEncryptor.new(current_user, user_session).perform
+      active_profile.recovery_code
```

Comment:
The only snag with this is that `Pii::ReEncryptor` no longer becomes an accurate class name. We would have to rename it to something like `ProfileRecoveryCodeGenerator`.